Cpp_test/lib/ipc-handler.cpp

```diff
@@ -2,90 +2,150 @@
-
-		} else {
-			if(pflag){
-				std::cout << "Pipe is used, pipe name is "<< info.method_name << std::endl;
-				std::cout << "File name: " << info.file_name << std::endl;
-				info.protocol = IPCProtocol::PIPE;	
-
-			} else if (qflag) {				
-				std::cout << "queue is used, queue name is "<< info.method_name << std::endl;
-				std::cout << "File name: " << info.file_name << std::endl;
-				info.protocol = IPCProtocol::QUEUE;
-
-			} else {
-				std::cout << "shared memory is used, shared name is "<< info.method_name << std::endl;
-				std::cout << "File name: " << info.file_name << std::endl;
-        		info.protocol = IPCProtocol::SHARE;
 
+	if (fflag)
+	{
+		if (pflag + qflag + sflag != 1)
+		{
+			if (info.file_name[0] == '-')
+			{
+				throw std::runtime_error("ERROR: File name can not start by '-', use -h for more information.\n");
 			}
-		}	
-
-	}else if (!hflag && (!pflag && !qflag && !sflag && !fflag)) {
+			else
+			{
+				std::cout << "name of file is used: " << info.file_name << std::endl;
+				std::cout << "Select one IPC method!e.g \n"
+							 "./ipcSend -q /<queue_name> -f <file_name>\n"
+							 "./ipcSend -p <pipe_name> -f <file_name>\n"
+							 "./ipcSend -s <shared_name> -f <file_name>\n";
+			}
+		}
+		else
+		{
+			if (pflag == 1)
+			{
+				if (info.method_name[0] == '-')
+				{
+					throw std::runtime_error("ERROR: Method name can not start by '-', use -h for more information.\n");
+				}
+				else
+				{
+					std::cout << "Pipe is used, pipe name is " << info.method_name << std::endl;
+					std::cout << "File name: " << info.file_name << std::endl;
+					info.protocol = IPCProtocol::PIPE;
+				}
+			}
+			else if (qflag == 1)
+			{
+				if (info.method_name[0] == '-')
+				{
+					throw std::runtime_error("ERROR: Method name can not start by '-', use -h for more information.\n");
+				}
+				else
+				{
+					std::cout << "queue is used, queue name is " << info.method_name << std::endl;
+					std::cout << "File name: " << info.file_name << std::endl;
+					info.protocol = IPCProtocol::QUEUE;
+				}
+			}
+			else if (sflag == 1)
+			{
+				if (info.method_name[0] == '-')
+				{
+					throw std::runtime_error("ERROR: Method name can not start by '-', use -h for more information.\n");
+				}
+				else
+				{
+					std::cout << "shared memory is used, shared name is " << info.method_name << std::endl;
+					std::cout << "File name: " << info.file_name << std::endl;
+					info.protocol = IPCProtocol::SHARE;
+				}
+			}
+		}
+	}
+	else if(qflag >= 2 || pflag >= 2 || sflag >= 2 )
+	{
+		throw std::runtime_error("ERROR: Method must be applied 1 time, use -h for more information.\n");
+	}
+	else if(qflag + pflag + sflag >= 2)
+	{
+		throw std::runtime_error("ERROR: Only use 1 method for each time, use -h for more information.\n");
+	}
+	else if((qflag ==1 &&info.method_name[0] == '-') || (pflag ==1 &&info.method_name[0] == '-') || (sflag ==1 &&info.method_name[0] == '-'))
```

Comment:
This condition still works, you can try with: ./ipc_send -s -methodName (or ./ipc_send --s -methodName, because I will change to use long option)

Reply:
Oh it's because you check the name of the method name only if the file is provided. In my opinion that is quite weird. Why are you checking the name of the method if the file is not provided?

I recommend you refactor your different tests. This is confusing.

Reply:
Well, it seems like not necessary to check this condition if the file is not given. I will remove it.